### PR TITLE
[SEMI-MODULAR]Require flavour text to join the round.

### LINKED ISSUE #9828: You can join rounds without flavor text
## Reproduction: Join a round without flavor text

https://i.imgur.com/BKd60GD.png

### PR DESCRIPTION
**Pull request by nevimer**
## About The Pull Request

Fixes https://github.com/Skyrat-SS13/Skyrat-tg/issues/9828
This will require characters to have flavour text to enter the round. 

https://user-images.githubusercontent.com/77420409/145351259-eda4a290-1d9d-4452-87af-9a858b4d9fbf.mp4

## How This Contributes To The Skyrat Roleplay Experience

More effort characters. 

## Changelog

:cl:

fix: Characters once again need flavour text to enter the round. 
/:cl:

### CODE REVIEW

File: code/controllers/subsystem/job.dm [hunk 1]
```diff
@@ -951,6 +951,11 @@ SUBSYSTEM_DEF(job)
 	if(possible_job.has_banned_species(player.client.prefs))
 		JobDebug("[debug_prefix] Error: [get_job_unavailable_error_message(JOB_UNAVAILABLE_SPECIES)], Player: [player][add_job_to_log ? ", Job: [possible_job]" : ""]")
 		return JOB_UNAVAILABLE_SPECIES
+	if(length(player.client.prefs.read_preference(/datum/preference/text/flavor_text)) <= 10)
```
Comment (Zonespace27) [inline code/controllers/subsystem/job.dm]: Maybe raise it above 10 characters? Feels extremely low, at the moment.

**Comment by Zonespace27** [comment]
Also: are ghost roles that spawn you as a new character (thus, flavor text-less) affected? I assume not, but just a sanity check just in case.

**Comment by nevimer** [comment]
> 

Nope, this purely affects round start and midround JOIN GAME based roles.

File: code/__DEFINES/jobs.dm [hunk 2]
```diff
@@ -6,7 +6,7 @@
 #define JOB_UNAVAILABLE_SLOTFULL 5
 /// Job unavailable due to incompatibility with an antag role.
 #define JOB_UNAVAILABLE_ANTAG_INCOMPAT 6
-
+#define JOB_UNAVAILABLE_FLAVOUR 7 // SKYRAT ADDITION
```
Comment (Seris02) [inline code/__DEFINES/jobs.dm]:
```suggestion
#define JOB_UNAVAILABLE_FLAVOUR 7 // SKYRAT EDIT ADDITION
```

File: code/controllers/subsystem/job.dm (hunk 1 above)
Comment (Seris02) [reply code/controllers/subsystem/job.dm]: This should be 150

**Comment by Seris02** [comment]
Needs something so that it actually informs the player that they can't join, rather than just show them that there aren't any jobs available, although that might be something I have to PR upstream honestly

**Comment by hugmeorelse** [comment]
Can you have this follow the character requirement we had? We basically required a minimum amount of characters, not counting whitespace.

**Comment by nevimer** [comment]
> 

Yeah, for sure. It's currently checking 150 characters of bytes at the proc.

**Pull request merged**